content/docs/guides/converting-gatsby-to-tina.mdx

```diff
@@ -343,22 +271,609 @@ Add the following to the top of `src/style.css`. This will resize any images in
+-      frontmatter {
+-        title
+-        date(formatString: "MMMM DD, YYYY")
+-        description
+-      }
+-    }
+-    previous: mdx(id: { eq: $previousPostId }) {
+-      fields {
+-        slug
+-      }
+-      frontmatter {
+-        title
+-      }
+-    }
+-    next: mdx(id: { eq: $nextPostId }) {
+-      fields {
+-        slug
+-      }
+      frontmatter {
+        title
+      }
+    }
+  }
+`
+```
+
+Now that we've configured our page with a new data source we can use th useTina hook to implement visual editing.
+
+first update the page props for `BlogPostTemplate`. We'll add in our server fetched data and pull that in using the useTina hook
+
+```diff
+
++ import { useTina } from 'tinacms/dist/react'
+//...
+
+- const BlogPostTemplate = ({
+- data: { previous, next, site, mdx: post },
+-  location,
+-  children,
+-}) => {
++ const BlogPostTemplate = ({
++   serverData,
++  data: { site },
++  location
++  }) => {
+
++ const { query, variables, nextPageData, previousPageData } = serverData
++ const { data: tinaData } = useTina({
++   data: serverData.data,
++   query,
++   variables,
++ })
+//...
+```
+
+Then we'll swap out all of the existing data with the data we get back from Tina. Note the additon of the tinaField property, which is used to add contextual editing for each of the fields.
+
+```diff
++ import {useTina } from 'tinacms/dist/react'
++ import { TinaMarkdown } from "tinacms/dist/rich-text";
+
+
+//...
+<header>
+-  <h1 itemProp="headline">{post.frontmatter.title}</h1>-
+-  </p>{post.frontmatter.date}<<p>
++  <h1 data-tina-field={tinaField(tinaData.post, 'title')} itemProp="headline">{tinaData.post.title}</h1>
++  <p data-tina-field={tinaField(tinaData.post, 'date')}>{tinaData.date}</p>
+</header>
+- {children}
++ <main data-tina-field={tinaField(tinaData.post, "body")}>
++   <TinaMarkdown content={tinaData.post.body} />
++ </main>
+<hr />
+<footer>
+  <Bio />
+</footer>
+</article>
+<nav className="blog-post-nav">
+<ul
+  style={{
+    display: `flex`,
+    flexWrap: `wrap`,
+    justifyContent: `space-between`,
+    listStyle: `none`,
+    padding: 0,}}>
+  <li>
+-  {previous && (
+-    <Link to={previous.fields.slug} rel="prev">
+-      ← {previous.frontmatter.title}
++  {previousPageData && (
++    <Link to={previousPageData.slug} rel="prev">
++      ← {previousPageData.title}
+        </Link>
+      )}
+    </li>
+    <li>
+
+- {next && (
+-   <Link to={next.fields.slug} rel="next">
+-                {next.frontmatter.title} →
++ {nextPageData && (
++   <Link to={nextPageData.slug} rel="next">
++               {nextPageData.title} →
+```
+
+There's one other step we'll do. Unfortunately, our date isn't being formatted using by the graphql query. To fix this we'll use a library to format our date.
+
+```bash
+yarn add dateformat
+```
+
+Then we'll add a useEffect statement to update the date when it date changes. We're using useEffect here so that the date will be recomputed when we use the visual editor.
```

Comment:
```suggestion
Then we'll add a `useEffect` to update the date when the date changes. We're using `useEffect` here so that the date will be recomputed when we use the visual editor.
```